### PR TITLE
Perf improvement

### PR DESCRIPTION
**Pull request by harshjain2**
Proposed changes that could improve perf.

### CODE REVIEW

File: src/Microsoft.TestPlatform.PlatformAbstractions/net46/Tracing/EqtTrace.cs [hunk 1]
```diff
@@ -28,6 +28,11 @@ public partial class PlatformEqtTrace : IPlatformEqtTrace
         /// <param name="childDomain">Child <c>AppDomain</c>.</param>
         public void SetupRemoteEqtTraceListeners(AppDomain childDomain)
         {
+            if(!isInitialized)
```
Comment (smadala) [inline src/Microsoft.TestPlatform.PlatformAbstractions/net46/Tracing/EqtTrace.cs]: As discussed with @AbhitejJohn , Verify that there is no side effects.

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/EqtTrace.cs [hunk 2]
```diff
@@ -164,13 +164,10 @@ public bool ShouldTrace(PlatformTraceLevel traceLevel)
...
                     return false;
                 case PlatformTraceLevel.Error:
-                    return Source.Switch.ShouldTrace(TraceEventType.Error);
                 case PlatformTraceLevel.Warning:
-                    return Source.Switch.ShouldTrace(TraceEventType.Warning);
                 case PlatformTraceLevel.Info:
-                    return Source.Switch.ShouldTrace(TraceEventType.Information);
                 case PlatformTraceLevel.Verbose:
-                    return Source.Switch.ShouldTrace(TraceEventType.Verbose);
+                    return isInitialized;
```
Comment (smadala) [inline src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/EqtTrace.cs]: Check isInitialized before calling `ShouldTrace()`.

**Review by smadala — CHANGES_REQUESTED** [review]

**Review by smadala — APPROVED** [review]

File: src/Microsoft.TestPlatform.PlatformAbstractions/net46/Tracing/EqtTrace.cs (hunk 1 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.PlatformAbstractions/net46/Tracing/EqtTrace.cs]: verified, no side effects :)

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/EqtTrace.cs (hunk 2 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/EqtTrace.cs]: ShouldTrace is a public API. And I feel the right place is this method only.

**Comment by harshjain2** [comment]
@dotnet-bot Windows_NT / Release Build

**Comment by harshjain2** [comment]
@dotnet-bot test Windows_NT / Release Build

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/EqtTrace.cs [hunk 3]
```diff
@@ -159,6 +159,11 @@ public bool InitializeVerboseTrace(string customLogFile)
         /// <inheritdoc/>
         public bool ShouldTrace(PlatformTraceLevel traceLevel)
         {
+            if(!isInitialized)
```
Comment (codito) [inline src/Microsoft.TestPlatform.PlatformAbstractions/common/Tracing/EqtTrace.cs]: Please validate https://github.com/Microsoft/vstest-docs/blob/master/docs/diagnose.md#collect-trace-using-config-file scenario

**Review by codito — APPROVED** [review]

**Comment by harshjain2** [comment]
@dotnet-bot test Windows_NT / Debug Build

**Pull request merged**